File: app/src/brainstorm/app/models/api/note.ts
import {
    FormBuilder,
    FormGroup,
    Validators
} from '@angular/forms';

import { EntityBase } from './entity-base';
import { Topic } from './topic';

export interface Note extends EntityBase {
    topicId: number;
    title: string;
    body: string;
    priority: number;

    topic?: Topic;
}

export const GenerateNoteForm = (note: Note, fb: FormBuilder): FormGroup =>
    fb.group({
        id: [note?.id],
        topicId: [note?.topicId],
        title: [note?.title, Validators.required],
        body: [note?.body],
        priority: [note?.priority, [ Validators.min(0), Validators.max(5) ]]
    });

File: server/Brainstorm.Models/Entities/Note.cs
namespace Brainstorm.Models.Entities;
public class Note : EntityBase
{
    public int TopicId { get; set; }
    public string Title { get; set; }
    public string Body { get; set; }
    public int Priority { get; set; }

    public Topic Topic { get; set; }

    public override void Settle() => Url = Encode(Title);
}